File: service-hello/src/main/java/com/acme/services/hello/java/HelloServiceImpl.java
package com.acme.services.hello.java;

import com.acme.contracts.services.hello.HelloService;
import com.acme.common.AbstractService;

public class HelloServiceImpl extends AbstractService implements HelloService {
    @Override
    public String sayHello(String name) {
        log("sayHello(\"" + name + "\")");

        return "Hello " + ((name == null) ? "World" : name);
    }

    @Override
    public String addEnthusiasm(String str) {
        log("addEnthusiasm(\"" + str + "\")");

        return str + "!";
    }
}

File: service-hello/src/main/kotlin/com/acme/services/hello/kotlin/HelloServiceImpl.kt
package com.acme.services.hello.kotlin

import com.acme.contracts.services.hello.HelloService
import com.acme.common.AbstractService.log

@Suppress("unused")
class HelloServiceImpl : HelloService {
    override fun sayHello(name: String): String {
        log("sayHello(\"$name\")")

        return "Hello $name"
    }

    override fun addEnthusiasm(str: String): String {
        log("addEnthusiasm(\"$str\")")

        return "$str!"
    }
}
